src/ziggurat/messaging/rabbitmq/producer.clj

```diff
@@ -24,19 +22,30 @@
       (assoc props :expiration (str expiration))
       props)))
 
+(defn- network-exception?
+  [e]
+  (let [exception-class (class e)]
+    (or (= com.rabbitmq.client.AlreadyClosedException exception-class) (= java.io.IOException exception-class))))
+
+(defn- publish-internal
+  [connection exchange message-payload expiration]
+  (try
+    (with-open [ch (lch/open connection)]
+      (lb/publish ch exchange "" (nippy/freeze (dissoc message-payload :headers))
+                  (properties-for-publish expiration (:headers message-payload))))
+    (catch Exception e
+      (log/error e "Exception was encountered while publishing to RabbitMQ")
+      (if (network-exception? e)
```

Comment:
why do you need to do this? can you just catch the exception in the catch expression?

Reply:
If the exception is raised due to an already closed RabbitMQ server, or an IO exception, it needs to return true and wait infinitely for the server to be up. So here we are identifying the error type.

Reply:
instead of returning true or false, can we return some kind of a keyword? @macalimlim LMK if that sounds correct. I think a return value is required here to terminate the recur condition, which is the caller of this function

Reply:
@WickedBrat I mean something like this `(... (catch AlreadyClosedException (... true) (catch IOException (... true)) (catch Exception (... false)))`

Reply:
@shubhang93 instead of booleans, what keywords do we have to return?

Reply:
Okay